media-api/app/controllers/MediaApi.scala

```diff
@@ -138,6 +147,17 @@ class MediaApi(
     }
   }
 
+  def diffProjection(id: String) = auth.async { request =>
+    val onBehalfOfFn: OnBehalfOfPrincipal = auth.getOnBehalfOfPrincipal(request.user)
+    getImageResponseFromES(id, request) flatMap {
+      case Some((source, _, _, _)) =>
+        gridClient.getImageLoaderProjection(id, onBehalfOfFn) map (
+        projection => respond(JsonDiff.diff(Json.toJson(source), Json.toJson(projection)))
+      )
+      case _ => Future.successful(ImageNotFound(id))
+    }
```

Comment:
How about this? Longer, but easier in my opinion to comprehend. Plus if works when there is no record in ES/S3. Which feels right.

```suggestion
    for {
      maybeEsImage <- getImageResponseFromES(id, request)
      maybeEsJson = maybeEsImage.map{ case (source, _, _, _) => Json.toJson(source) }
      maybeProjectedImage <- gridClient.getImageLoaderProjection(id, onBehalfOfFn)
      maybeProjectedJson = maybeProjectedImage.map(Json.toJson)
    } yield {
      (maybeEsJson, maybeProjectedJson) match {
        case (None, None) => ImageNotFound(id)
        case (es, projected) => respond(JsonDiff.diff(
          es.getOrElse(JsObject.empty),
          projected.getOrElse(JsObject.empty)
        ))
      }
    }
```